test/e2e/hsts_policy_test.go

```diff
@@ -0,0 +1,171 @@
+		MaxAge:                  maxAgePolicy,
+	}
+
+	ing := &configv1.Ingress{}
+	appsDomain := ""
+
+	// Update the ingress config with the new HSTS policy, and include its apps domain
+	if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		// Get the ingress config
+		if err := kclient.Get(context.TODO(), clusterConfigName, ing); err != nil {
+			t.Logf("Get ingress config failed: %v, retrying...", err)
+			return false, nil
+		}
+
+		// Update the hsts.DomainPatterns with the ingress spec domain
+		if !domainExists(ing.Spec.Domain, hstsPolicy.DomainPatterns) {
+			hstsPolicy.DomainPatterns = append(hstsPolicy.DomainPatterns, "*."+ing.Spec.Domain)
+			appsDomain = ing.Spec.Domain
+		}
+
+		// Update the ingress config with the new policy
+		ing.Spec.RequiredHSTSPolicies = append(ing.Spec.RequiredHSTSPolicies, hstsPolicy)
+		// Update the ingress config
+		if err := kclient.Update(context.TODO(), ing); err != nil {
+			t.Logf("failed to update ingress config: %v, retrying...", err)
+			return false, nil
+		}
+
+		return true, nil
+
+	}); err != nil {
+		t.Fatalf("failed to update ingress config: %v", err)
+	}
+	defer func() {
+		// Remove the HSTS policies from the ingress config
+		ing.Spec.RequiredHSTSPolicies = nil
+		if err := kclient.Update(context.TODO(), ing); err != nil {
+			t.Fatalf("failed to restore ingress config: %v", err)
+		}
+	}()
+
+	p := ing.Spec.RequiredHSTSPolicies[0]
+	t.Logf("created a RequiredHSTSPolicy with DomainPatterns: %v,\n preload policy: %s,\n includeSubDomains policy: %s,\n largest age: %d,\n smallest age: %d\n", p.DomainPatterns, p.PreloadPolicy, p.IncludeSubDomainsPolicy, *p.MaxAge.LargestMaxAge, *p.MaxAge.SmallestMaxAge)
+
+	// Use the same namespace for route, service, and pod
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hsts-policy-namespace",
+		},
+	}
+	if err := kclient.Create(context.TODO(), ns); err != nil {
+		t.Fatalf("failed to create namespace: %v", err)
+	}
+	defer func() {
+		// this will cleanup all components in this namespace
+		if err := kclient.Delete(context.TODO(), ns); err != nil {
+			t.Fatalf("failed to delete test namespace %s: %v", ns.Name, err)
+		}
+	}()
+
+	// Create pod
+	echoPod := buildEchoPod("hsts-policy-echo", ns.Name)
+	if err := kclient.Create(context.TODO(), echoPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", ns, echoPod.Name, err)
+	}
+
+	// Create service
+	echoService := buildEchoService(echoPod.Name, ns.Name, echoPod.ObjectMeta.Labels)
+	if err := kclient.Create(context.TODO(), echoService); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+	}
+
+	// Create a route that should fail the HSTS policy validation
+	t.Logf("creating an invalid route at %s", time.Now().Format(time.StampMilli))
+	invalidRoute := buildRouteWithHSTS(echoPod.Name+"2", echoPod.Namespace, echoService.Name, icName.Name+"."+appsDomain, "max-age=99999999")
```

Comment:
Maybe name the route "invalid-route" instead of concocting a name based on the pod's name. Ditto for the "valid" case. Given that we're creating in our own temporary namespace we know they won't clash.

Reply:
done